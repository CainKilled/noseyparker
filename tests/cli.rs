use assert_cmd::prelude::*;
// use assert_fs::prelude::*;
use insta::{assert_display_snapshot, assert_snapshot};
// use predicates::prelude::*;
use std::process::Command;

macro_rules! assert_cmd_snapshot {
    ( $cmd:expr ) => {
        let cmd = $cmd;
        let output = cmd.get_output();
        let status = output.status;
        assert_display_snapshot!(status);
        let stdout = String::from_utf8(output.stdout.clone()).unwrap();
        assert_snapshot!(stdout);
        let stderr = String::from_utf8(output.stderr.clone()).unwrap();
        assert_snapshot!(stderr);
    };
}

fn noseyparker() -> Command {
    Command::cargo_bin("noseyparker").expect("noseyparker should be executable")
}

#[test]
fn test_noseyparker_no_args() {
    assert_cmd_snapshot!(noseyparker().assert().failure());
}

#[test]
fn test_noseyparker_help() {
    assert_cmd_snapshot!(noseyparker().arg("help").assert().success());
}

#[test]
fn test_noseyparker_help_scan() {
    assert_cmd_snapshot!(noseyparker().args(&["help", "scan"]).assert().success());
}

#[test]
fn test_noseyparker_help_summarize() {
    assert_cmd_snapshot!(noseyparker().args(&["help", "summarize"]).assert().success());
}

#[test]
fn test_noseyparker_help_report() {
    assert_cmd_snapshot!(noseyparker().args(&["help", "report"]).assert().success());
}

#[test]
fn test_noseyparker_help_datastore() {
    assert_cmd_snapshot!(noseyparker().args(&["help", "datastore"]).assert().success());
}

#[test]
fn test_noseyparker_help_rules() {
    assert_cmd_snapshot!(noseyparker().args(&["help", "rules"]).assert().success());
}
