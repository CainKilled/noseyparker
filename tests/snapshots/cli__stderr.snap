---
source: tests/cli.rs
expression: "String::from_utf8(output.stderr.clone()).unwrap()"
---

